modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/appserver/integration/common/utils/CarbonCommandToolsUtil.java

```diff
@@ -0,0 +1,280 @@
+     * @throws XPathExpressionException
+     */
+    public static boolean isServerDown(AutomationContext automationContext, int portOffset)
+            throws XPathExpressionException, InterruptedException {
+        boolean isPortOpen = true;
+        long startTime = System.currentTimeMillis();
+        // Looping the isPortOpen method, waiting for a while  to check the server is down or not
+        while (isPortOpen && (System.currentTimeMillis() - startTime) < TIMEOUT) {
+            isPortOpen = ClientConnectionUtil.isPortOpen(
+                    Integer.parseInt(FrameworkConstants.SERVER_DEFAULT_HTTPS_PORT) + portOffset);
+            if(isPortOpen){
+                Thread.sleep(1000); // waiting 1 sec to check isPortOpen again
+            }
+        }
+        return !isPortOpen;
+    }
+
+    /**
+     * This method is to return operating system
+     *
+     * @return if current os is windows return true : else false
+     */
+    public static String getCurrentOperatingSystem() {
+        return System.getProperty(FrameworkConstants.SYSTEM_PROPERTY_OS_NAME).toLowerCase();
+    }
+
+    /**
+     * This method is to execute commands and reading the logs to find the expected string.
+     *
+     * @param directory      - Directory which has the file to be executed .
+     * @param cmdArray       - Command array to be executed.
+     * @param expectedString - Expected string in  the log.
+     * @return boolean - true : Found the expected string , false : not found the expected string.
+     * @throws IOException - Error while getting the command directory
+     */
+    public static boolean isScriptRunSuccessfully(String directory, String[] cmdArray,
+                                                  String expectedString) throws IOException {
+        boolean isFoundTheMessage = false;
+        BufferedReader br = null;
+        Process process = null;
+        try {
+            File commandDir = new File(directory);
+            process = Runtime.getRuntime().exec(cmdArray, null, commandDir);
+            String line;
+            long startTime = System.currentTimeMillis();
+            while ((System.currentTimeMillis() - startTime) < TIMEOUT) {
+                br = new BufferedReader(new InputStreamReader(process.getInputStream(), "UTF-8"));
+                while ((line = br.readLine()) != null) {
+                    log.info(line);
+                    if (line.contains(expectedString)) {
+                        log.info("found the string " + expectedString + " in line " + line);
+                        isFoundTheMessage = true;
+                        break;
+                    }
+                }
+                if (isFoundTheMessage) {
+                    break;
+                }
+            }
+            return isFoundTheMessage;
+        } catch (IOException ex) {
+            log.error("Error when reading the InputStream when running shell script  " +
```

Comment:
I think you don't need to append ex.getMessage() to the log message. since exception object is already there.